R/download_data_wrds_compustat.R

```diff
@@ -85,9 +89,44 @@ download_data_wrds_compustat <- function(type, start_date, end_date, additional_
         inv = at / at_lag - 1,
         inv = if_else(at_lag <= 0, NA, inv)
       )
+  }
+
+  if (grepl("compustat_quarterly", type, fixed = TRUE)) {
+    fundq_db <- tbl(con, in_schema("comp", "fundq"))
+
+    compustat <- fundq_db |>
+      filter(
+        indfmt == "INDL" &
+          datafmt == "STD" &
+          consol == "C" &
+          between(datadate, start_date, end_date)
+      ) |>
+      select(
+        gvkey, datadate, rdq, fqtr, fyearq,
+        all_of(additional_columns)
+      ) |>
+      collect()
 
     disconnection_connection(con)
 
-    processed_data
+    compustat <- compustat |>
+      drop_na(fqtr)|>
+      mutate(date = ceiling_date(datadate, "quarter") %m-% months(1),
+             timepoint = paste0(fyearq, fqtr)) |>
+      group_by(gvkey, timepoint) |>
+      filter(datadate == max(datadate)) |>
+      ungroup() |>
+      filter(if_else(is.na(rdq), TRUE, date < rdq)) |>
```

Comment:
What is "rdq" and why is it so important? Maybe add a comment to the code?

Reply:
It is the reporting date, i.e., the date the data was first made public. Obviously, it has to be after the accounting period end. It is a common filter.

I do not think we should include a comment as it is inconsistent with the rest of the package. We can discuss adding many comments though.

Reply:
There are a few comments if the logic is not obvious and that is desireable (we don't have to comment everything because the code should be clear). I think it makes sense to have one here because "rdq" does not say anything about its meaning. Same goes for this recoding missing values of Fama-French.